### PR TITLE
Fix mhlo.scatter implicit batch update dims

### PR DESCRIPTION
**Pull request by rsuderman**
The update dims should have been updated during the implicit batch dimension transform.

### CODE REVIEW

File: compiler/src/iree/compiler/InputConversion/MHLO/MHLOToMHLOPreprocessing.cpp [hunk 1]
```diff
@@ -1296,9 +1301,10 @@ struct MHLOToMHLOPreprocessingPass
     patterns.insert<ExpandRngNormal, MulCastOfBool>(context);
 
     // scatter canonicalization patterns
-    patterns.insert<ScatterOpImplicitIndex, ScatterOpImplicitBatch,
-                    ScatterMaterializeInsertedDim, ScatterOpCollapseBatch>(
-        context);
+    patterns.insert<ScatterOpImplicitIndex>(context);
```
Comment (jpienaar) [inline compiler/src/iree/compiler/InputConversion/MHLO/MHLOToMHLOPreprocessing.cpp:1304]: This change seems unrelated (the two forms here are equivalent)

File: compiler/src/iree/compiler/InputConversion/MHLO/MHLOToMHLOPreprocessing.cpp [hunk 2]
```diff
@@ -526,6 +526,11 @@ struct ScatterOpImplicitBatch : public OpRewritePattern<mhlo::ScatterOp> {
           op, "Unable to add implicit batch dim to indice.");
     }
 
+    llvm::SmallVector<int64_t> newUpdateWindowDims;
+    for (auto dim : dimNumbers.getUpdateWindowDims()) {
+      newUpdateWindowDims.push_back(dim + 1);
+    }
```
Comment (jpienaar) [inline compiler/src/iree/compiler/InputConversion/MHLO/MHLOToMHLOPreprocessing.cpp:532]: Why is it always +1? (I would have guessed it depends on where insertion happened)

**Review by jpienaar — APPROVED** [review]

File: compiler/src/iree/compiler/InputConversion/MHLO/MHLOToMHLOPreprocessing.cpp (hunk 1 above)
Comment (rsuderman) [reply compiler/src/iree/compiler/InputConversion/MHLO/MHLOToMHLOPreprocessing.cpp]: Oops, this was some debugging code to track down the offending pass. Reverted.

File: compiler/src/iree/compiler/InputConversion/MHLO/MHLOToMHLOPreprocessing.cpp (hunk 2 above)
Comment (rsuderman) [reply compiler/src/iree/compiler/InputConversion/MHLO/MHLOToMHLOPreprocessing.cpp:534]: Added a comment. The batch dimension is inserted at the first index so everything shifts forward by 1.
